### DISCUSSION TITLE
Quadratura pagamenti

### DISCUSSION THREAD

**Discussion by ilguardiano** [fattureincloud/api#429] [Q&A] [answered]
Finalmente sono riuscito a far funzionare l'sdk per l'integrazione su fatture incloud. 
Nello specifico io creo degli ordini che poi trasferisco a fatture in cloud.
C'è pero' un problema che non riesco a risolvere.
Io creo un ordine con una sola riga col grossamount = 100 ed un pagamento col amount = 100 e passa.
Creo un ordine con due righe con grossamount = 20 e gross amount = 70 ed un pagamento col amount = 90 e mi dice che il totale dei pagamenti non corrisponde al totale da pagare.
In pratica con un ordine di una riga funziona, di più righe no. Non capisco il perchè.

**Comment by ilguardiano** [comment]
Nessuno ha mai avuto lo stesso problema. Cioè come faccio ad avere più righe ed un unico pagamento? C'è qualche esempio dove posso curiosare?

**Reply by valmoz** [reply]
Buongiorno @ilguardiano 
ha provato a leggere questa guida? [Invoice Totals Calculation](https://developers.fattureincloud.it/docs/guides/invoice-totals)

Le consiglio anche di passarci il JSON (opportunamente modificato per nascondere i dati sensibili) con cui sta effettuando la richiesta, in modo tale da poter verificare il problema. 

In alternativa può scrivere al supporto clienti, in modo tale da permetterci di recuperare i log della sua richiesta tramite il company ID.

Grazie
Mauro

**Comment by ilguardiano** [comment]
Per la verità quella guida mi era sfuggita, ma, pur guardandola e cercando di fare gli arrotondamenti ai due decimali il risultato non cambia.
Ecco il mio Json
```
{
  "type": "order",
  "entity": {
    "type": "company",
    "id": 82446699,
    "code": "",
    "name": "XXXXXXXXXXX.",
    "vat_number": "XXXXXXXXXXX",
    "tax_code": "XXXXXXXXXXX",
    "address_city": "MILANO",
    "address_province": "MI",
    "country": "Italia"
  },
  "number": 19,
  "date": "2024-03-01",
  "items_list": [
    {
      "code": "24-0001",
      "name": "( - Modena-Ponte San Marco)",
      "description": "01/03/2024 DDT:293",
      "qty": 1.0,
      "net_price": 258.20,
      "gross_price": 315.000,
      "vat": {
        "id": 8287263,
        "value": 22.0,
        "description": "IVA al 22%",
        "is_disabled": false
      }
    },
    {
      "code": "24-0002",
      "name": "( - Modena-Ponte San Marco)",
      "description": "01/03/2024 DDT:295",
      "qty": 1.0,
      "net_price": 258.20,
      "gross_price": 315.000,
      "vat": {
        "id": 8287263,
        "value": 22.0,
        "description": "IVA al 22%",
        "is_disabled": false
      }
    }
  ],
  "payments_list": [
    {
      "status": "not_paid",
      "due_date": "2024-03-01",
      "amount": 630.000,
      "payment_terms": {
        "type": "standard",
        "days": 0
      }
    }
  ],
  "amount_due_discount": 0.0
}
```

e l'errore restituito è sempre lo stesso:
Error calling CreateIssuedDocument: {"error":{"message":"Il totale dei pagamenti non corrisponde al totale da pagare.","validation_result":null}}'

**Reply by valmoz** [reply, marked as answer]
Buongiorno @ilguardiano 
deve impostare **use_gross_prices** a true e passare solamente i lordi.
Inoltre dovrebbe usare due cifre decimali.

Grazie
Mauro

**Comment by ilguardiano** [comment]
Grazie. impostando use gross prices ora funziona.
